dask_cloudprovider/providers/azure/azureml.py

```diff
@@ -394,17 +394,19 @@ async def __create_cluster(self):
 
         self.__print_message("Waiting for scheduler node's IP")
         while (
-            run.get_status() != "Canceled"
-            and run.get_status() != "Failed"
+            run.status != "Canceled"
+            and run.status != "Failed"
             and "scheduler" not in run.get_metrics()
         ):
             print(".", end="")
             logger.info("Scheduler not ready")
             time.sleep(5)
 
-        if run.get_status() == "Canceled" or run.get_status() == "Failed":
-            logger.exception("Failed to start the AzureML cluster")
-            raise Exception("Failed to start the AzureML cluster.")
+        if run.status == "Canceled" or run.status == "Failed":
+            run_details = run.get_details()
+            error_msg = run_details["error"]
```

Comment:
use .get() so you don't throw a KeyError instead

Reply:
Good catch.